OrleansDashboard/Metrics/GrainProfilerFilter.cs

```diff
@@ -78,17 +79,22 @@ private void Track(IIncomingGrainCallContext context, Stopwatch stopwatch, bool
 
         private bool ShouldSkipProfiling(IIncomingGrainCallContext context)
         {
-            try
-            {
-                return
-                    context.Grain.GetType().GetCustomAttribute<NoProfilingAttribute>() != null ||
-                    context.ImplementationMethod?.GetCustomAttribute<NoProfilingAttribute>() != null;
-            }
-            catch (Exception ex)
+            var method = context.ImplementationMethod;
+
+            return shouldProfileCache.GetOrAdd(context.ImplementationMethod, _ =>
```

Comment:
I think there should be a check for `context.ImplementationMethod != null` before checking the dictionary

Reply:
Thanks, have improved it.